pkg/tpm/tcpa_log.go

```diff
@@ -0,0 +1,730 @@
+	eventInfo += gptEvent.uefiPartitionHeader.DiskGUID.String()
+	return &eventInfo, nil
+}
+
+func getImageLoadEventString(eventData []byte) (*string, error) {
+	var eventReader = bytes.NewReader(eventData)
+	var imageLoadEvent EFIImageLoadEvent
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &imageLoadEvent.imageLocationInMemory); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &imageLoadEvent.imageLengthInMemory); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &imageLoadEvent.imageLinkTimeAddress); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &imageLoadEvent.lengthOfDevicePath); err != nil {
+		return nil, err
+	}
+
+	// Stop here we only want to know which device was used here.
+
+	eventInfo := fmt.Sprintf("Image loaded at address 0x%d ", imageLoadEvent.imageLocationInMemory)
+	eventInfo += fmt.Sprintf("with %db", imageLoadEvent.imageLengthInMemory)
+
+	return &eventInfo, nil
+}
+
+func getVariableDataString(eventData []byte) (*string, error) {
+	var eventReader = bytes.NewReader(eventData)
+	var variableData EFIVariableData
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.variableName.blockA); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.variableName.blockB); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.variableName.blockC); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.variableName.blockD); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.variableName.blockE); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.unicodeNameLength); err != nil {
+		return nil, err
+	}
+
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.variableDataLength); err != nil {
+		return nil, err
+	}
+
+	variableData.unicodeName = make([]uint16, variableData.unicodeNameLength)
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.unicodeName); err != nil {
+		return nil, err
+	}
+
+	variableData.variableData = make([]byte, variableData.variableDataLength)
+	if err := binary.Read(eventReader, binary.LittleEndian, &variableData.variableData); err != nil {
+		return nil, err
+	}
+
+	guid := fmt.Sprintf("Variable - %x-%x-%x-%x-%x - ", variableData.variableName.blockA, variableData.variableName.blockB, variableData.variableName.blockC, variableData.variableName.blockD, variableData.variableName.blockE)
+	eventInfo := guid
+	utf16String := utf16.Decode(variableData.unicodeName)
+	eventInfo += fmt.Sprintf("%s", string(utf16String))
+
+	return &eventInfo, nil
+}
+
+func getEventDataString(eventType uint32, eventData []byte) (*string, error) {
```

Comment:
this should be a method on the EventType type

and just return string, no need for pointers (returning empty string should not be a performance hit you care about). (and if it is, we gotta talk.)